internal/poller/poller.go

```diff
@@ -19,8 +21,15 @@ type Poller struct {
 }
 
 func New(interval time.Duration, pollFunc func() (interface{}, error)) *Poller {
+	wrappedFn := func() (interface{}, error) {
+		defer func() {
+			panics.LogAndPanic(recover(), debug.Stack())
+		}()
+		return pollFunc()
+	}
```

Comment:
I think we need to implement this as part of the pollFunc that we pass. Reason being that if we catch it here the stacktrace won't point to what caused the panic.